repos/spack_repo/builtin/packages/dorado/package.py

```diff
@@ -39,13 +48,42 @@ class Dorado(CMakePackage, CudaPackage):
     conflicts("%gcc@:8", msg="Dorado requires at least gcc@9 to compile.")
     conflicts("%gcc@13:", msg="Dorado will not build with gcc@13 and newer.")
 
-    patch("cmake-htslib.patch")
-    patch("hdf5-libaec.patch")
+    patch("hdf5-libaec-0.5.3.patch", when="@:0.5.3")
+    patch("hdf5-libaec-0.6.3.patch", when="@0.6:0.9")
 
     def setup_build_environment(self, env: EnvironmentModifications) -> None:
         env.prepend_path("LD_LIBRARY_PATH", self.spec["libdeflate"].prefix.lib64)
         env.prepend_path("LIBRARY_PATH", self.spec["libdeflate"].prefix.lib64)
 
+    def patch(self):
+        with when("@:0.9"):  # 1.0 removes these lines from the CMakeLists.txt files
+            filter_file(
+                "add_dependencies(dorado_lib htslib_project)", "", "CMakeLists.txt", string=True
+            )
+            filter_file(
+                "add_dependencies(dorado_utils htslib_project)",
+                "",
+                "dorado/utils/CMakeLists.txt",
+                string=True,
+            )
+
+            with when("@0.9.5:0.9.6"):
```

Comment:
I'm not sure this syntax works here, at the very least this filter_file function is being called with 0.6.3 resulting in an error.
```suggestion
           if self.spec.satisfies("@0.9.5:0.9.6"):
```
probably replace all the 'with when's in this function.